source/by-role/developer/common/targeting.rst

```diff
@@ -639,3 +641,134 @@ in the ``Entertainment`` category page.
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Emulating authorizedRoles in XB
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Some projects use authentication such as SSO based on SAML, etc., in the delivery experience. Crafter Studio allows you
+to setup and test those rules through targeting. In this section, we'll take a look at an example of emulating
+authentication of users using ``authorizedRoles`` in XB for limiting access to articles in the Editorial blueprint.
+
+One of the ways to define security for a project whether it's screens or pages in a site, is to configure security rules.
+To configure security rules for our example, in Studio, click on |projectTools| -> ``Configuration``. Select ``Engine Project
+Configuration`` from the list, then add the following highlighted section to the config file to configure URL restrictions:
+
+.. code-block:: xml
+    :caption: *Engine Project Configuration - Editorial Blueprint*
+    :emphasize-lines: 3-10
+    :linenos:
+
+    <site>
+      <version>4.0.1</version>
+      <security>
+        <urlRestrictions>
+          <restriction>
+            <url>/articles/**</url>
+            <expression>isAuthenticated()</expression>
+          </restriction>
+        </urlRestrictions>
+      </security>
+    </site>
+
+The URL restriction configured above indicates that authenticated users can access all items under ``/articles`` in the
+project and are able to view the page.
+
+Next, let's configure Audience Targeting in XB which allows you to pick a role you want to emulate and see how your pages
+will react. To configure roles to emulate, in Studio, click on |projectTools| -> ``Configuration``. Select ``User
+Interface Configuration`` from the list, then add the following highlighted section to the config file:
+
+.. code-block:: xml
+    :caption: *User Interface Configuration*
+    :emphasize-lines: 11-32
+    :linenos:
+
+    ...
+    <widget id="craftercms.components.ToolsPanelPageButton">
+      <configuration>
+        <target id="icePanel"/>
+        <title id="previewAudiencesPanel.title" defaultMessage="Audience Targeting"/>
+        <icon id="@mui/icons-material/EmojiPeopleRounded"/>
+        <widgets>
+          <widget id="craftercms.components.PreviewAudiencesPanel">
+            <configuration>
+              <fields>
+                <roles>
+                  <id>roles</id>
+                  <name>User Role</name>
+                  <description>User Role</description>
+                  <type>dropdown</type>
+                  <defaultValue>anonymous</defaultValue>
+                  <values>
+                    <value>
+                      <label>Anonymous</label>
+                      <value>anonymous</value>
+                    </value>
+                    <value>
+                      <label>Role A</label>
+                      <value>RoleA</value>
+                    </value>
+                    <value>
+                      <label>Role B</label>
+                      <value>RoleB</value>
```

Comment:
It may not be part of this PR but one detail is that I don't see any mention to where these roles are defined. For example, is "anonymous" something understood by the system or is that defined somewhere? Or is it simply that it doesn't match one of the defined roles so it is essentially interpreted as anonymous?

Similarly for Role(A|B), are those defined somewhere?